src/App.tsx

```diff
@@ -1,14 +1,74 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { Option } from './types/Option';
+
+interface FilterOptions {
+  select: string;
+  query: string;
+}
+
+function filterBy(todos: Todo[], { select, query }: FilterOptions): Todo[] {
+  let filtered = todos;
+
+  if (select) {
+    switch (select) {
+      case Option.ACTIVE:
+        filtered = filtered.filter(todo => !todo.completed);
+        break;
+
+      case Option.COMPLETED:
+        filtered = filtered.filter(todo => todo.completed);
+        break;
+
+      default:
+        return filtered;
+    }
+  }
+
+  if (query) {
+    const normilizeQuery = query.toLowerCase().trim();
+
+    filtered = filtered
+      .filter(todo => todo.title.toLowerCase().includes(normilizeQuery));
+  }
+
+  return filtered;
+}
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [isVisibleModal, setIsVisibleModal] = useState(false);
+  const [select, setSelect] = useState('');
+  const [query, setQuery] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [post, setPost] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setIsLoading(false);
+    }, 300);
```

Comment:
Instead of it should set loading as a false after catch method in [finally method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally)